app/src/main/java/org/simple/clinic/scanid/ScanSimpleIdEffectHandler.kt

```diff
@@ -38,8 +38,25 @@ class ScanSimpleIdEffectHandler @AssistedInject constructor(
       .addConsumer(OpenPatientSummary::class.java, ::openPatientSummary, schedulersProvider.ui())
       .addConsumer(OpenPatientSearch::class.java, ::openPatientSearch, schedulersProvider.ui())
       .addTransformer(OnlinePatientLookupWithIdentifier::class.java, onlinePatientLookupWithIdentifier())
+      .addTransformer(SaveCompleteMedicalRecords::class.java, saveCompleteMedicalRecords())
       .build()
 
+  private fun saveCompleteMedicalRecords(): ObservableTransformer<SaveCompleteMedicalRecords, ScanSimpleIdEvent> {
+    return ObservableTransformer { effects ->
+      effects
+          .observeOn(schedulersProvider.io())
+          .map(SaveCompleteMedicalRecords::completeMedicalRecords)
+          .map { completeMedicalRecords ->
+            completeMedicalRecords to completeMedicalRecords.map { medicalRecord ->
+              patientRepository.saveCompleteMedicalRecord(medicalRecord)
+            }
+          }
```

Comment:
`.map` is the wrong operator to use here since we are not converting the medical records to anything. What we should be using is a `.doOnNext` instead.